File: simple/src/main/java/io/github/tuuzed/adapter/simple/MainActivity.java
package io.github.tuuzed.adapter.simple;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import io.github.tuuzed.adapter.Items;
import io.github.tuuzed.adapter.RecyclerViewAdapter;
import io.github.tuuzed.adapter.simple.provider.IntegerItemProvider;
import io.github.tuuzed.adapter.simple.provider.LongItemProvider;
import io.github.tuuzed.adapter.simple.provider.StringItemProvider;

public class MainActivity extends AppCompatActivity {

    private Items mItems;
    private RecyclerViewAdapter mAdapter;
    private RecyclerView mRecyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mRecyclerView = (RecyclerView) findViewById(R.id.recyclerView);
        mAdapter = new RecyclerViewAdapter(mItems = new Items());
        mAdapter.register(String.class, new StringItemProvider());
        mAdapter.register(Integer.class, new IntegerItemProvider());
        mAdapter.register(Long.class, new LongItemProvider());

        mRecyclerView.setLayoutManager(new LinearLayoutManager(this));
        mRecyclerView.setAdapter(mAdapter);
        for (int i = 0; i < 5; i++) {
            mItems.add((long) i);
            mItems.add(String.valueOf(i));
            mItems.add(i);
        }
        mAdapter.notifyDataSetChanged();
    }
}

File: sample/src/main/java/com/tuuzed/androidx/recyclerview/adapter/sample/MainActivity.kt
package com.tuuzed.androidx.recyclerview.adapter.sample

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.text.InputType
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.tuuzed.androidx.recyclerview.adapter.RecyclerViewAdapter
import com.tuuzed.androidx.recyclerview.adapter.prefs.prefDsl
import com.tuuzed.androidx.recyclerview.adapter.prefs.withPrefs
import kotlinx.android.synthetic.main.main_act.*

class MainActivity : AppCompatActivity() {

    private lateinit var cxt: Context
    private lateinit var listAdapter: RecyclerViewAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        cxt = this
        setContentView(R.layout.main_act)
        recyclerView.layoutManager = LinearLayoutManager(this)
        listAdapter = RecyclerViewAdapter.with(recyclerView).withPrefs().prefDsl {
            prefGeneral {
                title = "Feed List"
                summary = "Feed List"
                callback = { _, _ ->
                    startActivity(Intent(cxt, FeedListActivity::class.java))
                    true
                }
            }
            prefCategory {
                title = "Prefs"
            }
            prefGeneral {
                title = "General"
                summary = "General#summary"
            }
            prefEditText {
                title = "EditText"
                summary = "EditText#summary"
                inputType = InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_FLAG_MULTI_LINE
            }
            prefDivider {}
            prefCheckBox {
                title = "CheckBox"
                summary = "CheckBox#summary"
            }
            prefSwitch {
                title = "Switch"
                summary = "Switch#summary"
            }
            prefRadio {
                title = "Radio"
                summary = "Radio#summary"
            }
            prefDivider {}
            prefSingleList {
                title = "SingleList"
                summary = "SingleList#summary"
                itemsLoader = {
                    it(arrayListOf(
                            "Single Item 1",
                            "Single Item 2",
                            "Single Item 3",
                            "Single Item 4"
                    ))
                }
                itemsLoader = {
                    it(arrayListOf(
                            "Single Item 1",
                            "Single Item 2",
                            "Single Item 3",
                            "Single Item 4"
                    ))
                }
            }
            prefMultiList {
                title = "MultiList"
                summary = "MultiList#summary"
                itemsLoader = {
                    it(arrayListOf(
                            "Multi Item 1",
                            "Multi Item 2",
                            "Multi Item 3",
                            "Multi Item 4"
                    ))
                }
                itemsLoader = {
                    it(arrayListOf(
                            "Multi Item 1",
                            "Multi Item 2",
                            "Multi Item 3",
                            "Multi Item 4"
                    ))
                }
            }
        }
        listAdapter.notifyDataSetChanged()
    }
}
